esp-radio/src/common_adapter.rs

```diff
@@ -350,66 +323,75 @@ type ModemClockControllerPeripheral = esp_hal::peripherals::WIFI<'static>;
-    const CALIBRATION_MODE: esp_phy_calibration_mode_t =
-        esp_wifi_sys::include::esp_phy_calibration_mode_t_PHY_RF_CAL_FULL;
-    #[cfg(not(phy_full_calibration))]
-    const CALIBRATION_MODE: esp_phy_calibration_mode_t =
-        esp_wifi_sys::include::esp_phy_calibration_mode_t_PHY_RF_CAL_PARTIAL;
-
-    #[cfg(phy_skip_calibration_after_deep_sleep)]
-    let calibration_mode = if crate::hal::system::reset_reason()
-        == Some(crate::hal::rtc_cntl::SocResetReason::CoreDeepSleep)
+pub(crate) fn enable_wifi_power_domain() {
+    #[cfg(not(any(soc_has_pmu, esp32c2)))]
     {
-        esp_wifi_sys::include::esp_phy_calibration_mode_t_PHY_RF_CAL_NONE
-    } else {
-        CALIBRATION_MODE
-    };
-    #[cfg(not(phy_skip_calibration_after_deep_sleep))]
-    let calibration_mode = CALIBRATION_MODE;
+        cfg_if::cfg_if! {
+            if #[cfg(soc_has_lpwr)] {
+                let rtc_cntl = esp_hal::peripherals::LPWR::regs();
+            } else {
+                let rtc_cntl = esp_hal::peripherals::RTC_CNTL::regs();
+            }
+        }
+        cfg_if::cfg_if! {
+            if #[cfg(esp32)] {
+                // The ESP32 doesn't require this.
+            } else if #[cfg(soc_has_apb_ctrl)] {
+                let syscon = esp_hal::peripherals::APB_CTRL::regs();
+            } else {
+                let syscon = esp_hal::peripherals::SYSCON::regs();
+            }
+        }
 
-    debug!("Using calibration mode {}", calibration_mode);
+        rtc_cntl
+            .dig_pwc()
+            .modify(|_, w| w.wifi_force_pd().clear_bit());
 
-    let res = CAL_DATA.with(|cal_data| unsafe {
-        register_chipv7_phy(
-            init_data,
-            cal_data as *mut _ as *mut crate::binary::include::esp_phy_calibration_data_t,
-            calibration_mode,
-        )
-    });
+        #[cfg(not(esp32))]
+        unsafe {
+            const WIFIBB_RST: u32 = 1 << 0; // Wi-Fi baseband
+            const FE_RST: u32 = 1 << 1; // RF Frontend RST
+            const WIFIMAC_RST: u32 = 1 << 2; // Wi-Fi MAC 
+
+            const BTBB_RST: u32 = 1 << 3; // Bluetooth Baseband
+            const BTMAC_RST: u32 = 1 << 4; // deprecated
+            const RW_BTMAC_RST: u32 = 1 << 9; // Bluetooth MAC
+            const RW_BTMAC_REG_RST: u32 = 1 << 11; // Bluetooth MAC Regsiters
+            const BTBB_REG_RST: u32 = 1 << 13; // Bluetooth Baseband Registers
+
+            const MODEM_RESET_FIELD_WHEN_PU: u32 = WIFIBB_RST
+                | FE_RST
+                | WIFIMAC_RST
+                | if cfg!(soc_has_bt) {
+                    BTBB_RST | BTMAC_RST | RW_BTMAC_RST | RW_BTMAC_REG_RST | BTBB_RST
+                } else {
+                    0
+                };
+
+            syscon
+                .wifi_rst_en()
+                .modify(|r, w| w.bits(r.bits() | MODEM_RESET_FIELD_WHEN_PU));
+            syscon
+                .wifi_rst_en()
+                .modify(|r, w| w.bits(r.bits() & !MODEM_RESET_FIELD_WHEN_PU));
```

Comment:
Want to save an instruction or two? This could be `read-write-write` instead of `modify-modify` :)